src/main/java/io/github/jhipster/registry/config/JWTSecurityConfiguration.java

```diff
@@ -4,32 +4,85 @@
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
 import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
 
 @Configuration
 @EnableWebSecurity
-@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
 @Profile("!" + Constants.PROFILE_OAUTH2)
 public class JWTSecurityConfiguration extends WebSecurityConfigurerAdapter {
 
+    private final AuthenticationManagerBuilder authenticationManagerBuilder;
+
     private final Http401UnauthorizedEntryPoint authenticationEntryPoint;
 
     private final TokenProvider tokenProvider;
 
-    public JWTSecurityConfiguration(Http401UnauthorizedEntryPoint authenticationEntryPoint,
-                                 TokenProvider tokenProvider) {
+    private final String username;
+
+    private final String password;
+
+    private final String[] roles;
+
+    public JWTSecurityConfiguration(@Value("${spring.security.user.name}") String username,
+                                    @Value("${spring.security.user.password}") String password,
+                                    @Value("${spring.security.user.roles}") String[] roles,
+                                    AuthenticationManagerBuilder authenticationManagerBuilder,
+                                    Http401UnauthorizedEntryPoint authenticationEntryPoint,
+                                    TokenProvider tokenProvider) {
+        this.username = username;
+        this.password = password;
+        this.roles = roles;
+        this.authenticationManagerBuilder = authenticationManagerBuilder;
         this.authenticationEntryPoint = authenticationEntryPoint;
         this.tokenProvider = tokenProvider;
     }
 
+    @Bean
+    public AuthenticationManager authenticationManager() {
+        try {
+            return authenticationManagerBuilder
+                .userDetailsService(userDetailsService())
+                .passwordEncoder(passwordEncoder())
+                .and()
+                .build();
+        } catch (Exception e) {
+            throw new BeanInitializationException("Security configuration failed", e);
+        }
+    }
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return new BCryptPasswordEncoder();
+    }
+
+    @Bean
+    public UserDetailsService userDetailsService() {
+        InMemoryUserDetailsManager manager = new InMemoryUserDetailsManager();
+        manager.createUser(
+            User.withUsername(username)
+                .password(passwordEncoder().encode(password))
```

Comment:
the password needs to be encode, otherwise, the authenticationManager won't allow the user to log in.
how about if the password is already encoded in property? So this code won't probably work